src/PageContent/Calendar/ContactContent.js

```diff
@@ -0,0 +1,124 @@
+import { AccountType, StringFormat } from '../../Enums';
+import './Content.css';
+import phone from '../../icons/phone.svg';
+
+class DescriptionContent extends Component {
+    constructor(props) {
+        super(props);
+        if (
+            this.props.delivery.donatingAgency &&
+            this.props.accountType === AccountType.DONATING_AGENCY_MEMBER
+        ) {
+            this.state = {
+                edit: false,
+                name: this.props.delivery.donatingAgency.contact.name,
+                phone: this.props.delivery.donatingAgency.contact.phone,
+                email: this.props.delivery.donatingAgency.contact.email
+            };
+        } else if (
+            this.props.delivery.receivingAgency &&
+            this.props.accountType === AccountType.RECEIVING_AGENCY
+        ) {
+            this.state = {
+                edit: false,
+                name: this.props.delivery.receivingAgency.contact.name,
+                phone: this.props.delivery.receivingAgency.contact.phone,
+                email: this.props.delivery.receivingAgency.contact.email
+            };
+        } else {
+            this.state = {
+                edit: false
+            };
+        }
+
+        this.edit = this.edit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+    }
+    edit() {
+        this.setState({
+            edit: true
+        });
+    }
+
+    handleChange(e) {
+        e.preventDefault();
+        this.setState({
+            name: e.target.name.value,
+            phone: e.target.phone.value,
+            edit: false
+        });
+    }
+
+    render() {
+        return (
+            <div className="wrapper">
+                <img className="content-icon" src={phone} alt="volunteer" />
+                <div className="content-wrapper content-wrapper-description">
+                    {this.props.accountType === AccountType.RECEIVING_AGENCY ? (
+                        <h1 className="section-header">
+                            Primary Contact for Delivery
+                        </h1>
+                    ) : this.props.accountType ===
+                    AccountType.DONATING_AGENCY_MEMBER ? (
+                            <h1 className="section-header">
+                            Primary Contact Pick Up
+                            </h1>
+                        ) : null}
+                    {!this.state.edit ? (
+                        <div className="content-details-wrapper">
+                            <p className="content-details contact-content">
+                                {this.state.name} ({this.state.phone})
+                            </p>
+                        </div>
+                    ) : (
+                        <div className="content-details-wrapper">
+                            <form
+                                className="edit-dg"
+                                onSubmit={this.handleChange}
+                            >
```

Comment:
There needs to be two different modes of edit for DA vs RA. For DAs, it needs to be a dropdown list for them to choose one of the member accounts (instead of directly modifying the name, phone, etc). For RAs, it's the text input fields you have now.

To make the dropdown list, first mock dummy data in `componentDidMount` in this component:
```
componentDidMount() {
    // backend TODO: fetch member list from db
    let memberList = [
        {
            uid: 'uid1',
            name: 'member1',
            phone: '111-111-1111',
            email: 'member1@test.com'
        },
        {
            uid: 'uid2',
            name: 'member2',
            phone: '222-222-2222',
            email: 'member2@test.com'
        }
    ];
}
```
then you can do something like in `RecurringPickupRequest.js`:
```
<select name="primaryContact" defaultValue="" required>
    <option value="" disabled>Select</option>
    {this.state.memberList.map((member,i) => {
        return (
            <option key={i} value={i}>{member.name}</option>
        );
    })}
</select>
```

Reply:
totally missed this sorry!